src/number_format.js

```diff
@@ -728,14 +732,14 @@ class NumberFormat extends React.Component {
 
   onBlur(e: SyntheticInputEvent) {
     const {props, state} = this;
-    const {format, onBlur} = props;
+    const {format, onBlur, allowLeadingZeros} = props;
     let {numAsString} = state;
     const lastValue = state.value;
 
     this.focusedElm = null;
 
     if (!format) {
-      numAsString = fixLeadingZero(numAsString);
+      numAsString = allowLeadingZeros ? numAsString : fixLeadingZero(numAsString);
```

Comment:
Change it to 
```js
if (!allowLeadingZeros) {
   numAsString = fixLeadingZero(numAsString);
}
```
This has less logical complexity with if is 1, while with ternary it's 2 (ternary is if + else). 
Can be on the same line as well.

Reply:
updated